### PR TITLE
Resolve half of #67 (Part 1/2)

### LINKED ISSUE #67: Crates API and documentation
Both crates basically lacks documentation, and <./rust-k256> lacks API choices.

Also their `name` are ...temporary.

PS It would be nice to think about traits used and reexported so that it would not clash with versions that downstream use. Though most likely Cargo will do its job on it, just it's better to be sure. 
This one is more about releasing the crates when things will come to it.

### PR DESCRIPTION
**Pull request by skaunov**
`arkworks-rs` have some issue while publishing (to <crates.io>) concerned with `serde` dependency. I don't feel like delving into that one is reasonable giving that it's outdated anyway. It's API have been sorted out, and documented.

The other crate can be found at <https://crates.io/crates/plume_rustcrypto>. Any changes to it are welcome: let's add it to this PR. 

Then squash and merge.

Note from Aayush: We still need to update plume_arkworks, add simple examples, and fix the docs URLs.

### CODE REVIEW

File: rust-arkworks/src/lib.rs [hunk 1]
```diff
@@ -1,10 +1,26 @@
-use crate::error::HashToCurveError;
+/// This crate provides the PLUME signature scheme.
+///
+/// See <https://blog.aayushg.com/nullifier> for more information.
+/// 
+/// Find RustCrypto crate as `plume_crypto`.
```
[the remark this answers is not in the document]
Comment (Divide-By-0) [reply rust-arkworks/src/lib.rs:5]: This one can be plume_arkworks

File: rust-k256/Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,12 @@
 [package]
-name = "zk-nullifier"
+name = "plume_rustcrypto"
```
Comment (Divide-By-0) [inline rust-k256/Cargo.toml:2]: Let's make this one plume_crypto, and the other one plume_arkworks!

File: rust-k256/src/lib.rs [hunk 3]
```diff
@@ -1,101 +1,69 @@
 // #![feature(generic_const_expr)]
 // #![allow(incomplete_features)]
 
+//! A library for generating (coming [soon](https://github.com/plume-sig/zk-nullifier-sig/issues/84)) and verifying PLUME signatures.
+//!
+//! See <https://blog.aayushg.com/nullifier> for more information.
+//!
+//! Find `arkworks-rs` crate as `plume_arkworks`.
```
Comment (Divide-By-0) [inline rust-k256/src/lib.rs:8]: Ah you already did this in some places!

**Review by Divide-By-0 — CHANGES_REQUESTED** [review]
Mostly on crate rename.

**Comment by skaunov** [comment]
@Divide-By-0 ,
that `plume_rustcrypto` naming was actually last minute change, since I too wanted to go with "plume_crypto" for some time. But going through all the thing and some crates I felt like it's not communicating that clear that there's two versions: for `arkworks-rs` and for RustCrypto ecosystems. So made it a bit more specific. Let's give some time to make its way to how they both look... And it can be transferred easily to that other name. :shrug:

**Comment by Divide-By-0** [comment]
> RustCrypto

That makes sense then, as long as the naming is consistent i.e. fix the 'Find RustCrypto crate as `plume_crypto`.' comment!

**Comment by skaunov** [comment]
All are very valid points, I just need to come to something on how to best express them. :sweat_smile: I mean some doesn't go that smooth when trying.

1) I don't like this *bare* look too. Though all I came up with is the link to this repo <README.MD> file; since that page isn't documentation yet, but it contains the link to the documentation (the one which `cargo doc` / `pub` interfaces) in the designated section. When it's not a monorepo stuff we have in the actual <README.MD> goes there. Hence the link. I don't like copying it, since it's kinda awkward to bump version with just <README.MD> changes... Any better ideas are welcome!

2) That one is confusing, since there's a couple of quick ways to get to <https://docs.rs/plume_rustcrypto/> really fast, and the link from the crate page is one of them. So people are really don't expect that.

3) This one is really needed. I thought that the API is really small for adding an example, but they're always good! The problem is that without `sign` method it's hard to show the thing concisely; so I guess I'll do that one before closing this. Also I'll add the example to the other crate just so it would be present and maintained, and came naturally when that crate will be there.

For the last one I added a HAZMAT notice.

**Comment by Divide-By-0** [comment]
> All are very valid points, I just need to come to something on how to best express them. 😅 I mean some doesn't go that smooth when trying.
> 
> 1. I don't like this _bare_ look too. Though all I came up with is the link to this repo <README.MD> file; since that page isn't documentation yet, but it contains the link to the documentation (the one which `cargo doc` / `pub` interfaces) in the designated section. When it's not a monorepo stuff we have in the actual <README.MD> goes there. Hence the link. I don't like copying it, since it's kinda awkward to bump version with just <README.MD> changes... Any better ideas are welcome!
> 2. That one is confusing, since there's a couple of quick ways to get to https://docs.rs/plume_rustcrypto/ really fast, and the link from the crate page is one of them. So people are really don't expect that.
> 3. This one is really needed. I thought that the API is really small for adding an example, but they're always good! The problem is that without `sign` method it's hard to show the thing concisely; so I guess I'll do that one before closing this. Also I'll add the example to the other crate just so it would be present and maintained, and came naturally when that crate will be there.
> 
> For the last one I added a HAZMAT notice.

1. Sweet. It's totally fine to bump version with a README change -- no one's using it yet haha.
2. Yeah, we should just include that docs.rs link in the README and in the github repo.
3. Makes sense, lets do it in that order then.

**Review by Divide-By-0 — APPROVED** [review]

**Comment by Divide-By-0** [comment]
Going to merge this for now so that the README shows up in the crates docs -- i expect this issue is still open however till we get sign merged etc!

**Comment by skaunov** [comment]
Two tiny notes. I meant not the version counter but the fact that it 
would be always behind actual <README.MD> without any reason for it. 
And basically the link is always on the page where people used to find 
it; no problem to double it, but it would be helpful only for that rare 
occasion when someone would receive the crate outside <crates.io> and 
Github. %)

**Pull request merged**